//
// Copyright 2019 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package asylo.sgx;

import "asylo/crypto/certificate.proto";
import "asylo/identity/provisioning/sgx/internal/tcb.proto";

// A set of SGX Provisioning Certification Key (PCK) certificates for
// all configured TCB levels of an SGX-enabled platform.
message PckCertificates {
  // A combination of PCK certificate and its corresponding TCB level.
  message PckCertificateInfo {
    // Required.
    optional asylo.sgx.Tcb tcb_level = 1;

    // Required.
    optional asylo.sgx.RawTcb tcbm = 2;

    // The certificate must be in X.509 PEM format. Required.
    optional asylo.Certificate cert = 3;
  }

  // The order of elements does not matter.
  repeated PckCertificateInfo certs = 1;
}
